pkg/endpoint/endpoint.go

```diff
@@ -360,14 +381,125 @@ func (e *Endpoint) WaitForProxyCompletions() error {
 	return nil
 }
 
+// runK8sCiliumEndpointSync starts a controller that syncronises the endpoint
+// to the corresponding k8s CiliumEndpoint CRD
+// ep.Mutex must be help for the setup portion of this function, but not the
+// controller.
+func (e *Endpoint) runK8sCiliumEndpointSync() {
+	elog := e.getLogger()
+	endpointID := e.ID
+
+	controllerName := fmt.Sprintf("sync-to-k8s-ciliumendpoint (%v)", endpointID)
+	epName := fmt.Sprintf("%v-cep-%v", node.GetName(), endpointID)
+
+	if !k8s.IsEnabled() {
+		elog.WithField("name", controllerName).Debug("Not starting controller because k8s is disabled")
+		return
+	}
+
+	ciliumClient, err := getCiliumClient()
+	if err != nil {
+		elog.WithError(err).Error("Not starting controller because unable to get cilium k8s client")
+		return
+	}
+
+	var lastMdl *models.Endpoint
+	mdlEqual := func(a, b *models.Endpoint) bool {
+		if a == nil || b == nil {
+			return false
+		}
+
+		aJSON, err := a.MarshalBinary()
+		if err != nil {
+			return false
+		}
+
+		bJSON, err := b.MarshalBinary()
+		if err != nil {
+			return false
+		}
+
+		return string(aJSON) == string(bJSON)
+	}
+
+	// NOTE: The controller functions do NOT hold the endpoint locks
+	e.controllers.UpdateController(controllerName,
+		controller.ControllerParams{
+			RunInterval: time.Duration(5) * time.Second,
+			DoFunc: func() error {
+				scopedLog := elog.WithField("controller", controllerName)
+
+				mdl := e.GetModel()
+				if mdlEqual(mdl, lastMdl) {
+					scopedLog.Debug("Skipping CiliumEndpoint update because it has not changed")
+					return nil
+				}
+				lastMdl = mdl
+
+				namespace := e.GetK8sNamespace()
+				cep, err := ciliumClient.CiliumEndpoints(namespace).Get(epName, meta_v1.GetOptions{})
+				switch {
+				// A real error
+				case err != nil && !k8serrors.IsNotFound(err):
+					scopedLog.WithError(err).Error("Cannot get CEP for update")
+					return err
+
+				// do an update
+				case err == nil:
+					// Update the copy of the cep
+					(*cilium_v2.CiliumEndpointDetail)(mdl).DeepCopyInto(&cep.Status)
+
+					if _, err = ciliumClient.CiliumEndpoints(namespace).Update(cep); err != nil {
+						scopedLog.WithError(err).Error("Cannot update CEP")
+						return err
+					}
+
+					return nil
+				}
+
+				// The CEP was not found, this is the first creation of the endpoint
+				cep = &cilium_v2.CiliumEndpoint{
+					ObjectMeta: meta_v1.ObjectMeta{
+						Name: epName,
+					},
+					Status: cilium_v2.CiliumEndpointDetail(*mdl),
+				}
+
+				_, err = ciliumClient.CiliumEndpoints(namespace).Create(cep)
+				if err != nil {
+					scopedLog.WithError(err).Error("Cannot create CEP")
+					return err
+				}
+
+				return nil
+			},
+			StopFunc: func() error {
+				scopedLog := elog.WithField("controller", controllerName)
+
+				namespace := e.GetK8sNamespace()
+				if err := ciliumClient.CiliumEndpoints(namespace).Delete(epName, &meta_v1.DeleteOptions{}); err != nil {
+					scopedLog.WithError(err).Error("Unable to delete CEP")
```

Comment:
add the endpoint's name and namespace to the log message

Reply:
it should already be there, I think. `scopedLog` is derived from the endpoint logger